### DISCUSSION TITLE
substitutions json ?

### DISCUSSION THREAD

**Discussion by pquiring** [oracle/graal#3409] [Q&A] [answered]
What is the general format of SubstitutionsConfig.json for native-image?

Just interested in substituting a method in a class.

I know how to use @TargetClass but the json might be easier.

Thanks,

**Comment by olpaw** [comment, marked as answer]
Hi @pquiring 
The non-API `SubstitutionsConfig.json` format is a leftover from the past. It's implemented in `com.oracle.svm.hosted.substitute.DeclarativeSubstitutionProcessor`. Except for cosmetic changes it hasn't been updated for a while. 
Our plan is to remove it at some point. Do not rely on it. It's mostly unmaintained.

**Reply by pquiring** [reply]
Understood, thanks.
